uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/MsalTestApp.java

```diff
@@ -0,0 +1,128 @@
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.app;
+
+import androidx.annotation.NonNull;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiScrollable;
+import androidx.test.uiautomator.UiSelector;
+import com.microsoft.identity.client.ui.automation.installer.LocalApkInstaller;
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.MicrosoftStsPromptHandler;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.MicrosoftStsPromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A model for interacting with the Msal Test App for MSAL Android during UI Test.
+ */
+public class MsalTestApp extends App{
+
+    private final static String TAG = MsalTestApp.class.getSimpleName();
+    private static final String MSAL_TEST_APP_PACKAGE_NAME = "com.msft.identity.client.sample.local";
+    private static final String MSAL_TEST_APP_NAME = "MSAL Test App";
+    public static final String MSAL_TEST_APP_APK = "MsalTestApp.apk";
+    public static final String OLD_MSAL_TEST_APP_APK = "OldMsalTestApp.apk";
+
+    // constructors
+    public MsalTestApp() {
+        super(MSAL_TEST_APP_PACKAGE_NAME, MSAL_TEST_APP_NAME, new LocalApkInstaller());
+        localApkFileName = MSAL_TEST_APP_APK;
+    }
+
+    public MsalTestApp(final boolean installOldApk) {
+        super(MSAL_TEST_APP_PACKAGE_NAME, MSAL_TEST_APP_NAME, new LocalApkInstaller());
+        if (installOldApk) {
+            localApkFileName = OLD_MSAL_TEST_APP_APK;
+        } else {
+            localApkFileName = MSAL_TEST_APP_APK;
+        }
+        localUpdateApkFileName = MSAL_TEST_APP_APK;
+    }
```

Comment:
So when would this boolean be false? Same question to @mohitSinglaMS for OneAuthTestApp.